docs/docs/configuration/internal-format.md

```diff
@@ -0,0 +1,33 @@
+# Message internal format
+
+Hermes appends internal metadata to every published message. They are not visible to neither publisher nor subscriber,
+but are present on Kafka. Thus if you use tools to read data directly from Kafka, you will see events along with their
+metadata. Metadata format depends on data format (Avro, JSON) - please see format documentation for details.
+
+Currently metadata holds two fields:
+
+* **id** - string containing Hermes-Message-Id as UUID v4
+* **timestamp** - Unix epoch timestamp in milliseconds holding date of reception
+
+There is no guarantee that this metadata won't be extended.
+
+## JSON
+
+Messages with metadata have following format:
+
+```json
+{
+    "metadata": {
+        "id": "b0838e5f-c3d1-47b3-8949-abe7a1f65202",
+        "timestamp": 1439129947000
+    },
+    "message": {
+        // contents of your message goes here
+    }
+}
+```
+
+## Avro
+
+Hermes metadata is stored in `__metadata` field, which is sepcified as map of optional elements. It will always contain
```

Comment:
this should go into `publishing-avro` document, is this really the schema we ask our users to add? i would like to see nicer example as part of docs there. they are marked as incubating anyways, so not final.